src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -227,6 +232,19 @@ public sealed class ChainedHeaderTree : IChainedHeaderTree
             this.UnconsumedBlocksCount = 0;
         }
 
+        /// <inheritdoc />
+        public ChainedHeaderTreeState State()
+        {
+            var state = new ChainedHeaderTreeState
+            {
+                ChainedHeadersByHash = this.chainedHeadersByHash,
```

Comment:
We probably actually want to create copies of each collection no? (or the intention is to get the changes as they are reflected on the tree)

Also these methods should be internal so they will not get abused.

Reply:
@dangershony these methods can't be internal as they sit on the IConsensusManager interface ;) We need public access to them.

I will make copies of the collections though 👍